### PR TITLE
Convert `ipaddresses` to 4x API namespaced function

### LINKED ISSUE #270: Suggestion: Rename ipaddresses function and namespace it for this ssh module
Hi saz!  I just ran into a somewhat unrelated issue where a function, deep_merge, was implemented in the simplib module as well as stdlib.  Puppet 5 didn't care, Puppet 6 flipped out.  In doing some investigation, I noticed that they also implemented ipaddresses as a function but it was implemented differently.  (more importantly, it didn't include the ability to exclude interfaces)  I've talked with them already and they are planning on updating all of theirs to be namespaced to simplib, but while I'm thinking about it, I wanted to throw that out as a suggestion here too.  Maybe namespace it as ssh::ipaddresses?  That would help prevent it from conflicting with others -- especially if someday puppet decided to include that in core or stdlib did.  (not especially likely, but still)

### PR DESCRIPTION
**Pull request by alexjfisher**
The modern API ruby functions don't have any environment isolation
issues, can be namespaced, perform automatic type checking etc.

Fixes #270

See
http://puppet-on-the-edge.blogspot.com/2015/01/the-puppet-4x-function-api.html
and
https://github.com/puppetlabs/puppet-specifications/blob/0286d30dc903f36c36879cef6f09a9226c8096fc/language/func-api.md

### CODE REVIEW

File: lib/puppet/parser/functions/ipaddresses.rb [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-    # always exclude loopback interface
-    interfaces_exclude = ['lo']
-    if args.size == 1
-      unless args[0].is_a?(Array)
-        raise(Puppet::ParseError, 'ipaddresses(): Requires first argument to be an Array')
-      end
-      interfaces_exclude << args[0]
-    end
-
-    return false unless networking.include?('interfaces')
```
Comment (alexjfisher) [inline lib/puppet/parser/functions/ipaddresses.rb]: I left this out as I couldn't figure out how the `networking` fact could exist, but it not have an `interfaces` key.

File: metadata.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.24.0 < 6.0.0"
+      "version_requirement": ">= 4.25.0 < 6.0.0"
```
Comment (alexjfisher) [inline metadata.json]: For type `Stdlib::IP::Address`

**Comment by bastelfreak** [comment]
@alexjfisher can you please rebase?

**Comment by alexjfisher** [comment]
rebased.

**Comment by bastelfreak** [comment]
@saz can you please merge this and do a new release afterwards?

**Pull request merged**